common/src/test/java/com/graphicsfuzz/common/util/SideEffectCheckerTest.java

```diff
@@ -62,4 +71,30 @@ public void testCountingLoopHasSideEffects() throws Exception {
         ShadingLanguageVersion.ESSL_310));
   }
 
+  // TODO(521): Unignore when #521 is ready to merge.
+  @Ignore
+  @Test
+  public void testOutParamHasSideEffects() throws Exception {
+    final String shader = "void main { "
+        + "   uint out1;"
+        + "   uvec2 out2;"
+        + "   uvec3 out3;"
+        + "   uvec4 out4;"
+        + "   uaddCarry(uint(0), uint(0), out1);"
+        + "   uaddCarry(uvec2(0, 0), uvec2(0, 0), out2);"
+        + "   uaddCarry(uvec3(0, 0, 0), uvec3(0, 0, 0), out3);"
+        + "   uaddCarry(uvec4(0, 0, 0, 0), uvec4(0, 0, 0, 0), out4);"
+        + "}";
+
+    final TranslationUnit tu = ParseHelper.parse(shader, ShaderKind.FRAGMENT);
+    tu.setShadingLanguageVersion(ShadingLanguageVersion.ESSL_310);
+
+    assertFalse(new CheckPredicateVisitor() {
```

Comment:
I like what you've done here, asserting regarding the name and side-effects each time a call using the built-in is reached.  Now that I see it I realise there's no point using a CheckPredicate visitor.  Please remove the outer assert and change it to:

new StandardVisitor() {
  // what you already have
}.visit(tu);